src/reference.cr

```diff
@@ -133,14 +133,50 @@ class Reference
     io << '>'
   end
 
+  # Returns whether this can be recursive or not.
+  # It calculates on compile-time, and it determines `exec_recursive` handling is needed really.
+  private def possibly_recursive?
```

Comment:
Structs can also be recursive...

Reply:
@asterite When structs are recursive, inner struct must be wrapped by reference type. So this is no problem.

Reply:
Ah, you are right. I guess the check will be done for any referenced types, but it doesn't make sense to do it for a struct.